dvd/dvd_dump.ixx

```diff
@@ -696,27 +778,44 @@ export bool redumper_dump_dvd(Context &ctx, const Options &options, DumpMode dum
 
     SignalINT signal;
 
-    uint8_t ss_range_idx = 0;
+    uint8_t skip_range_idx = 0;
+    bool kreon_locked = false;
     for(uint32_t s = 0; s < sectors_count;)
     {
         bool increment = true;
 
         uint32_t sectors_to_read = std::min(sectors_at_once, sectors_count - s);
 
-        // Skip Xbox Security Sector Ranges
-        if(is_xbox && ss_range_idx < xbox_ss_ranges.size())
+        // check if drive needs locking
+        if(s < xbox_lock_sector && s + sectors_to_read >= xbox_lock_sector)
+            sectors_to_read = std::min(sectors_to_read, xbox_lock_sector - s);
+        else if(s == xbox_lock_sector)
+        {
+            status = cmd_kreon_set_lock_state(*ctx.sptd, KREON_LockState::LOCKED);
+            if(status.status_code)
+                throw_line("failed to set lock state, SCSI ({})", SPTD::StatusMessage(status));
+            if(options.verbose)
+                LOG_R("locking kreon drive, sector: {}", s);
+            kreon_locked = true;
+        }
+
+        // skip xbox security sector ranges and L1 filler range
+        if(is_xbox && !kreon_locked && skip_range_idx < xbox_skip_ranges.size())
         {
-            // If in SS range then write zeroes until end of region
-            if(xbox_ss_ranges[ss_range_idx][0] <= s && s <= xbox_ss_ranges[ss_range_idx][1] + 1)
+            if(xbox_skip_ranges[skip_range_idx][0] <= s && s <= xbox_skip_ranges[skip_range_idx][1] + 1)
             {
-                if(s == xbox_ss_ranges[ss_range_idx][1] + 1)
+                if(s == xbox_skip_ranges[skip_range_idx][1] + 1)
                 {
-                    ss_range_idx++;
+                    if(options.verbose)
+                        LOG_R("skipped sectors: {}-{}", xbox_skip_ranges[skip_range_idx][0], xbox_skip_ranges[skip_range_idx][1]);
+                    skip_range_idx++;
+                    while(s > xbox_skip_ranges[skip_range_idx][1] + 1)
```

Comment:
Was there a reason to add this? Wouldn't you only hit this if there were overlapping regions?

Reply:
For xbox360, the second security sector range seems to exist wholly within the L1 filler. Since I'm adding the L1 filler as a 'skip range', I do need to do this.